nasa/ow_simulator
# Oceanwater 602 Faults Stop Arm for  Unstow and Stow

## Linked Issues:
| Jira Ticket 🎟️   | [Oceanwater-602](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-602) |
| ----------- | ----------- |
| EPIC ⚡| [Oceanwater-551](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-551) |
| Github :octocat:  | na |


## Summary of Changes
* if arm fault, stops stow and unstow movement from executing
* if no arm fault, the stow and unstow can be called
** the stopping of arms through faults can be called prior to action and also during action

## Test
* call any roslaunch 
* go into rqt and select any arm fault
* open a new terminal and try calling the rosservice for stow and unstow
* should receive the following and no movement should be seen in gazebo or rviz 

```
success: False
message: "Unstow action failed"
```
* uncheck the arm fault in dynamic reconfigure
*  call rosservice for stow/unstow and see execution as nominal
